### DISCUSSION TITLE
Is it possible to get and edit the current filename with IDEAVim's VimScript?

### DISCUSSION THREAD

**Discussion by mbwgh** [JetBrains/ideavim#703] [Q&A]
I would like to write a short utility function which allows to jump quickly between files differing only in their file extension.
I.e. between `.h` and `.c`, or between `.ts`, and `.html`.

Maybe even something more general like jumping between between `MyClass.java` and `MyClassDto.java` or something.

I noticed however, that the `%` and `#` registers are not supported. I opened an issue for this, because I didn't not find one that addressed this.  `#` is however supported as a special-case argument to `:e`.

Likewise, VimScript lacks functions like `expand`, or `:h filename-modifiers` like `:t` or `:p`.

So, is this (barely) possible in "IdeaVimScript", or is there some workaround, like using a different language or invoking Intellij commands something else?

**Comment by nozzlegear** [comment]
I'd also like to know this. I'm trying to write a small vimscript function that runs different types of Rider formatting actions based on the filetype, along with some custom cleanup once it's finished. But I can't find any way to get the filetype or filename -- `&ft` and `%` seem to be missing in IdeaVim.

**Comment by halilim** [comment]
I'm using a workaround by abusing the clipboard like:

```vim
function! FilePath()
  let l:original_clipboard = @+
  exec 'action CopyPathFromRepositoryRootProvider'
  let l:file_path = @+
  let @+ = original_clipboard
  return file_path
endfunction
```

For other path action types:

1. Action menu > IdeaVim: Track Action IDs
2. Action menu > Copy Path/Reference

Side effect: status bar displays "... has been copied".
